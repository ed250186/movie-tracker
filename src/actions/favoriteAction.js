export const addFavoriteMovie = (favorites, userId) => ({
    type: 'ADD_FAVORITE_MOVIE',
    favorites,
    userId
})

export const getFavoriteMovies = (favorites) => ({
    type: 'GET_FAVORITE_MOVIES',
    favorites
})

export const deleteFavoriteMovie = (favorites) => ({
    type: 'DELETE_FAVORITE_MOVIE',
    favorites
})