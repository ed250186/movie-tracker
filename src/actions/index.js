export const setMovies = (movies) => ({
    type: 'SET_MOVIES',
    movies
});

export const grabUsers = (users) => ({
    type: 'GRAB_USERS',
    users
});