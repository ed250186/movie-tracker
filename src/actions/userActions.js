export const grabUsers = grabUsers => ({
  type: "GRAB_USERS",
  grabUsers
});

export const addUsers = addUsers => ({
    type: "ADD_USERS",
    addUsers
});

export const signInUser = signInUser => ({
  type: "SIGNIN_USER",
  signInUser
});

export const signOutUser = signOutUser => ({
  type: "SIGNOUT_USER",
  signOutUser
});

export const signInError = error => ({
  type: "SIGNIN_ERROR",
  error
});

export const signUpError = error => ({
  type: "SIGNUP_ERROR",
  error
})
