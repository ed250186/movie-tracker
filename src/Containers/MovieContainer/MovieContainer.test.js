import {
  MovieContainer,
  mapStateToProps,
  mapDispatchToProps
} from "./MovieContainer.jsx";
import { setMovies } from "../../actions";
import {shallow} from 'enzyme';

describe("MovieContainer", () => {

  let wrapper;
    beforeEach(() => {
      wrapper = shallow(<MovieContainer/>)
    })

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  describe("mapStateToProps", () => {
    it("should return an object with a movie array", () => {
      const mockState = {
        movies: "Batman",
        type: "SET_MOVIES"
      };
      const expected = { movies: { movies: "Batman", type: "SET_MOVIES" } };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe("mapDispatchToProps", () => {
    it("calls dispatch with a setMovies action", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setMovies("Batman");
      const mappedDispatch = mapDispatchToProps(mockDispatch);
      mappedDispatch.setMovies("Batman");
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });


});
