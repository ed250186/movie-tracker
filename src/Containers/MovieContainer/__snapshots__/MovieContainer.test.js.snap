// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match snapshot 1`] = `<section />`;
